docs: describe how to manage embedded plugin registry for VS Code editor (#2503)
codenvy/che-docs: modules/administration-guide/partials/proc_removing-extensions-from-the-embedded-open-vsx-registry.adoc

[id="removing-extensions-from-the-embedded-open-vsx-registry"]
= Removing extensions from the embedded Open VSX Registry

> This section describes how to remove VS Code extensions from the embedded Open VSX Registry.

Reviewer: ```suggestion
This section describes how to remove Visual Studio Code extensions from the embedded Open VSX Registry.
```

.Procedure

> . Modify the plug-in registry's link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[`openvsx-sync.json`] file to add an extension:

Reviewer: ```suggestion
. Modify the plugin registry's link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[`openvsx-sync.json`] file to add an extension:
```

> . Modify `openvsx-sync.json` file in the link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[plugin-registry repository] to remove extensions from the list.

Reviewer: ```suggestion
. Modify the plug-in registry's link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[`openvsx-sync.json`] file to add an extension:
```

Reviewer: Note that this is a link to the downstream file... we probably need a new antora variable for this URL so we can pull either https://github.com/eclipse-che/che-plugin-registry/blob/main/openvsx-sync.json or https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json. 

Or should we link to the raw file instead?

https://raw.githubusercontent.com/eclipse-che/che-plugin-registry/main/openvsx-sync.json
vs.
https://raw.githubusercontent.com/redhat-developer/devspaces/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json

Reviewer 2: @nickboldt, I've already taken care of this by using snippet files in the examples folder. See MR # 22 in https://issues.redhat.com/browse/RHDEVDOCS-4761.

> . Modify `openvsx-sync.json` file in the link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[plugin-registry repository] to remove extensions from the list.

Reviewer: ```suggestion
. Modify the plug-in registry's link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[`openvsx-sync.json`] file to add an extension:
```

> . To build the plug-in registry container image and publish it to a container registry, run the following commands:

Reviewer: ```suggestion
. To build the plugin registry container image and publish it to a container registry, run the following commands:
```

> . Build the container image with plugin registry and push the image to a container registry:

Reviewer: ```suggestion
. To build the plug-in registry container image and publish it to a container registry, run the following commands:
```

> . Build the container image with plugin registry and push the image to a container registry:

Reviewer: ```suggestion
. To build the plug-in registry container image and publish it to a container registry, run the following commands:
```

+
[subs="+attributes,+quotes"]
----

> $ ./build.sh -o username -r quay.io -t next

Reviewer: ```suggestion
$ ./build.sh -o username -r quay.io -t custom
```

> $ {docker-cli} push username/che-theia-devspaces:next

Reviewer: ```suggestion
$ {docker-cli} push quay.io/username/che-plugin-registry:custom
```

----
. Customize an instance of `CheCluster` Custom Resource in the cluster to point the image and save the changes:
+
[source,yaml,subs="+quotes"]
----
spec:
  components:
    pluginRegistry:
      deployment:
        containers:

>           - image: quay.io/username/pluginregistry-rhel8:next

Reviewer: ```suggestion
          - image: quay.io/username/plugin-registry:custom
```

      openVSXURL: '' 
----

> . Check that `che-plugin-registry` pod restarted and it's running. Restart the workspace and check plugins in the Plugins view.

Reviewer: ```suggestion
. Check that `plugin-registry` pod restarted and is running. Restart the workspace and check the available extensions in the Extensions view.
```
